src/aten/sycl/EmbeddingBagKernel.h

```diff
@@ -0,0 +1,591 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <ATen/core/Array.h>
+
+#include <aten/sycl/BatchKernel.h>
+#include <aten/sycl/NumericLimits.h>
+#include <comm/SYCLContext.h>
+
+namespace at::native::xpu {
+
+namespace {
+
+constexpr int MODE_SUM = 0;
+constexpr int MODE_MEAN = 1;
+constexpr int MODE_MAX = 2;
+
+template <
+    typename scalar_t,
+    typename accscalar_t,
+    typename index_t,
+    int mode,
+    int vec_size,
+    typename vec_t,
+    typename vec_acc_t,
+    typename vec_idx_t>
+struct EmbeddingBagKernelFunctor {
```

Comment:
It should be the interface header of kernels. Please remove functor definition to EmbeddingBag.cpp